src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const extraData = action.extraData;
+
+      for (const key in extraData) {
+        state[key] = extraData[key];
+      }
+    } else if (action.type === 'removeProperties') {
+      const keysToRemove = action.keysToRemove;
 
+      for (const key of keysToRemove) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
+      }
+    } else if (action.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
```

Comment:
- please use the switch case statement here
- move 'addProperties', 'removeProperties', 'clear' to the constants as it will make a code cleaner